### ISSUE TITLE
Easy way to check available datasets

### ISSUE THREAD

**Issue by magdalenafuentes** [mir-dataset-loaders/mirdata#166]
Could be some kind of `mirdata.info` with a flag that discriminates datasets with available audio and those which haven't?

**Comment by lostanlen** [comment]
Having a list of available loaders for the current version is relatively easy:

```python
def info():
    exclude = ["download_utils", "jams_utils", "utils"]
    mirdata_submodules = []
    for module in sorted(list(sys.modules.keys())):
        split = module.split(".")
        if len(split) == 2 and split[0] == "mirdata" and split[1] not in exclude:
            mirdata_submodules.append(split[1])
    return mirdata_submodules
```

Knowing which ones have available audio is far more difficult. Of course, if the user supplies a `data_home`, then it's simply a matter of looking for subfolders and validating them.

However, what if the user has installed various datasets at various places during various sessions? Then mirdata needs a runtime configuration file (`.mirdatarc`) which keeps track of those. This would be a maintenance nightmare, to say nothing of thread safety, so i'm in favor of having `mirdata.info` be specific to a user-provided `data_home` argument.

Does this sound like a plan?

**Comment by magdalenafuentes** [comment]
Great suggestion! 

About the audio available, I meant to include a flag somewhere so before downloading any dataset, you can check if it has audio or not.

**Comment by lostanlen** [comment]
Ah, i see. Sorry, i misunderstood. Here's an updated version, where i check for the presence of the regexp `r'AUDIO_[A-Z]*[_]?REMOTE'` in the module.
This matches guitarset and medley_solos_db (and eventually tinysol).


```
def info(require_audio=False):
    exclude = ["download_utils", "jams_utils", "utils"]
    mirdata_submodules = []
    for module in sorted(list(sys.modules.keys())):
        split = module.split(".")
        if len(split) == 2 and split[0] == "mirdata" and split[1] not in exclude:
            if require_audio:
                audio_remotes = [
                    x_dir for x_dir in dir(sys.modules[module])
                    if re.match(r'AUDIO_[A-Z]*[_]?REMOTE', x_dir)]
                has_audio = len(audio_remotes)>0
                if has_audio:
                    mirdata_submodules.append(split[1])
            else:
                mirdata_submodules.append(split[1])
    return mirdata_submodules
```

I hope this helps!

**Comment by rabitt** [comment]
I guess there are really three states of data availability in mirdata - (1) you can download everything directly through mirdata (2) you can download some parts of the dataset through mirdata and other parts you have to find elsewhere (3) you cannot download anything through mirdata.

Maybe this is where a simple list on the front page of the readme with some check boxes could be useful? Maybe an extension of the ismir/mir_datasets table?

**Comment by lostanlen** [comment]
I see. @rabitt: for case (2), are you thinking about `beatles`?

> Maybe this is where a simple list on the front page of the readme with some check boxes could be useful?

Yes, that would be very convenient. Difficult to scale to 10+ datasets though. Perhaps we could have a (JSON?) file for every dataset which specifies these sort of things? So that we can format it and pretty-print it the way we want in the README, but also access it programmatically from inside the module.


> Maybe an extension of the ismir/mir_datasets table?

Sure. At the moment, there is only one column named "has audio?" (yes/no). From your experience, it seems like the situation is more nuanced.

**Comment by rabitt** [comment]
> I see. @rabitt: for case (2), are you thinking about `beatles`?

Yes exactly


> Yes, that would be very convenient. Difficult to scale to 10+ datasets though. Perhaps we could have a (JSON?) file for every dataset which specifies these sort of things? So that we can format it and pretty-print it the way we want in the README, but also access it programmatically from inside the module.

I like the idea of a json file with this info - it would also make it easy to merge our data with the data on the ismir_datasets page, as long as the datasets use the same key.

> Sure. At the moment, there is only one column named "has audio?" (yes/no). From your experience, it seems like the situation is more nuanced.

I think "has audio" or not is a bit too limiting, especially in cases where you might have partial audio, or no audio at all in the dataset itself (e.g. the MSD just has metadata). Instead I'd suggest having a column indicating what you can get from the downloader, i.e. "complete" data, "partial" data, or "unavailabe" data. As an optional second field we can put a short description of what "partial" data is missing/available.

**Comment by rabitt** [comment]
Will be fixed by #169 and #177
